app/core/models.py

```diff
@@ -329,3 +333,42 @@ class Sdg(AbstractBaseModel):
 
     def __str__(self):
         return f"{self.name}"
+
+
+class Affiliation(AbstractBaseModel):
+    """
+    Sponsor/partner relationships stored in this table are project-dependent.
+    They cannot be both a sponsor and a partner for the same project,
+    so if is_sponsor is partner they are a project partner,
+    if is_sponsor is sponsor they are a project sponsor.
+    """
+
+    class AffiliationChoices(Enum):
+        SPONSOR = "sponsor"
+        PARTNER = "partner"
+
+        def __str__(self):
+            return self.value
+
+        @classmethod
+        def choices(cls):
+            return [(key.value, key.name) for key in cls]
+
+    affiliate = models.ForeignKey(Affiliate, on_delete=models.CASCADE)
+    project = models.ForeignKey(Project, on_delete=models.CASCADE)
+    ended_at = models.DateTimeField("Ended at", null=True, blank=True)
+    is_sponsor = models.CharField(max_length=20, choices=AffiliationChoices.choices())
```

Comment:
This should be called something like `affiliation_type` so that code can do
```
if self.affiliation_type == "sponsor"
```
in line 371